src/Compute/Compute/VirtualMachine/Operation/GetAzureVMBootDiagnosticsDataCommand.cs

```diff
@@ -133,18 +135,19 @@ public override void ExecuteCmdlet()
                 if (this.Windows.IsPresent
                     || (this.Linux.IsPresent && !string.IsNullOrEmpty(this.LocalPath)))
                 {
-                    var screenshotUri = new Uri(result.Body.InstanceView.BootDiagnostics.ConsoleScreenshotBlobUri);
-                    var localFile = this.LocalPath + screenshotUri.Segments[2];
-                    DownloadFromBlobUri(screenshotUri, localFile);
+                    var bootDiagnostics = this.VirtualMachineClient.RetrieveBootDiagnosticsData(this.ResourceGroupName, this.Name);
+                    var localPathTest = this.LocalPath;
+                    var localFile = this.LocalPath + new Uri(bootDiagnostics.ConsoleScreenshotBlobUri).Segments[2];
+                    
+                    DownloadFromBlobUri(new Uri(bootDiagnostics.ConsoleScreenshotBlobUri), localFile);
                 }
 
 
                 if (this.Linux.IsPresent)
                 {
-                    var logUri = new Uri(result.Body.InstanceView.BootDiagnostics.SerialConsoleLogBlobUri);
-
+                    var bootDiagnostics = this.VirtualMachineClient.RetrieveBootDiagnosticsData(this.ResourceGroupName, this.Name);
+                    var logUri = new Uri(bootDiagnostics.SerialConsoleLogBlobUri);
```

Comment:
Serial console logs are available to download for windows too. This needs to be updated as Azure supports Screenshot & Serial logs for both Windows & Linux VMs.

Reply:
I edited it so Console and Serial logs are always downloaded for Windows.